azurerm/resource_arm_automation_schedule.go

```diff
@@ -0,0 +1,199 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/automation"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationScheduleCreateUpdate,
+		Read:   resourceArmAutomationScheduleRead,
+		Update: resourceArmAutomationScheduleCreateUpdate,
+		Delete: resourceArmAutomationScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"start_time": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateStartTime,
+			},
+
+			"expiry_time": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"frequency": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(automation.Day),
+					string(automation.Hour),
+					string(automation.Month),
+					string(automation.OneTime),
+					string(automation.Week),
+				}, true),
+			},
+			"timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func validateStartTime(v interface{}, k string) (ws []string, errors []error) {
+	starttime, tperr := time.Parse(time.RFC3339, v.(string))
+	if tperr != nil {
+		errors = append(errors, fmt.Errorf("Cannot parse %q", k))
+	}
+
+	u := time.Until(starttime)
+	if u < 5*time.Minute {
+		errors = append(errors, fmt.Errorf("%q should be at least 5 minutes in the future", k))
+	}
+
+	return
+}
+
+func resourceArmAutomationScheduleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationScheduleClient
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Schedule creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	accName := d.Get("account_name").(string)
+	freqstr := d.Get("frequency").(string)
+	freq := automation.ScheduleFrequency(freqstr)
+
+	cst := d.Get("start_time").(string)
+	starttime, tperr := time.Parse(time.RFC3339, cst)
+	if tperr != nil {
+		return fmt.Errorf("Cannot parse start_time: %s", cst)
```

Comment:
can we update this formatting argument to be `%q` to quote it?

Reply:
Fixed!